dali/pipeline/data/tensor_vector.cc

```diff
@@ -256,33 +249,56 @@ void TensorVector<Backend>::SetContiguous(bool contiguous) {
   }
 }
 
+
 template <typename Backend>
 void TensorVector<Backend>::Reset() {
+  tensors_.clear();
+  curr_tensors_size_ = 0;
+  type_ = {};
   if (IsContiguous()) {
-    type_ = {};
-    tensors_.resize(0);
     views_count_ = 0;
     tl_->Reset();
-  } else {
-    type_ = {};
-    for (auto &t : tensors_) {
-      t->Reset();
-    }
   }
 }
 
+
+template <typename Backend>
+template <typename SrcBackend>
+void TensorVector<Backend>::Copy(const TensorList<SrcBackend> &in_tl, cudaStream_t stream) {
+  SetContiguous(true);
+  tl_->Copy(in_tl, stream);
+
+  resize_tensors(tl_->ntensor());
+  views_count_ = 0;
```

Comment:
I don't think it's OK to reset it by hand.

Reply:
In fact, UpdateViews does that. I removed redundant `views_count_=0` calls. The remained ones are I guess necessary: in `Reset` and in `ShareData(TensorVector)`, which doesn't do UpdateViews (for obvious reasons).